src/clj/game/cards/ice.clj

```diff
@@ -2104,6 +2104,38 @@
                    :async true
                    :effect (req (do-access state :runner eid [:rd] {:no-root true}))}]}
 
+   "Slot Machine"
+   {:implementation "Encounter effect is manual"
+    :abilities [{:label "Roll them bones"
+                 :effect (effect (move :runner (first (:deck runner)) :deck)
+                                 (system-msg (str "uses Slot Machine to put the top card of the stack to the bottom,"
+                                                  " then reveal the top 3 cards in the stack: "
+                                                  (join ", " (map :title (take 3 (:deck runner)))))))}]
+    :subroutines [{:label "Runner loses 3 [credits]"
+                   :msg "Runner loses 3 [credits]"
+                   :effect (effect (lose-credits :runner 3))}
+                  {:label "Corp gains 3 [credits]"
+                   :effect (req (let [f (:type (nth (:deck runner) 0))
```

Comment:
I feel there is a nicer way of doing this involving list expansion and `take`.
E.g. `[unique-types (count (into #{} (map :type (take 3 (:deck runner)))))]`
Or as a threading macro: `[unique-types (-> (:deck runner) (take 3) (map :type) (into #{}) count)]`

And then checking if unique-subtypes <= 2 and == 1.

Reply:
Oh that's kind of cool. I like that. I'll add it in!